web/src/main/react-dashboard/src/components/TestRunView.js

```diff
@@ -49,563 +42,570 @@ import {FAIL, SUCCESS, ERROR, PENDING, RUNNING, HTTP_UNAUTHORIZED, LOGIN_URI, TE
+            + "?truncate=" + true;
 
-    if (this.props.active.reducer.currentInfra.testPlanStatus === FAIL ||
-      this.props.active.reducer.currentInfra.testPlanStatus === SUCCESS) {
-      fetch(logTruncatedContentUrl, {
-        method: "GET",
-        credentials: 'same-origin',
-        headers: {
-          'Accept': 'application/json'
-        }
-      })
-      .then(this.handleError)
-      .then(response => {
-        this.setState({
-          logDownloadStatus: response.ok ? SUCCESS : ERROR
-        });
-        return response.json();
-      }).then(data =>
-        this.setState({
-          logContent: data.inputStreamContent,
-          isLogTruncated: data.truncated,
-          inputStreamSize: data.completeInputStreamSize
+        fetch(testScenarioSummaryUrl, {
+            method: "GET",
+            credentials: 'same-origin',
+            headers: {
+                'Accept': 'application/json'
+            }
+        })
+            .then(this.handleError)
+            .then(response => {
+                this.setState({
+                    testSummaryLoadStatus: response.ok ? SUCCESS : ERROR
+                });
+                return response.json();
+            }).then(data => this.setState({
+            testScenarioSummaries: data.scenarioSummaries,
+            scenarioTestCaseEntries: data.scenarioTestCaseEntries
         }));
+
+        if (currentInfra.testPlanStatus === FAIL || currentInfra.testPlanStatus === SUCCESS) {
+            fetch(logTruncatedContentUrl, {
+                method: "GET",
+                credentials: 'same-origin',
+                headers: {
+                    'Accept': 'application/json'
+                }
+            })
+                .then(this.handleError)
+                .then(response => {
+                    this.setState({
+                        logDownloadStatus: response.ok ? SUCCESS : ERROR
+                    });
+                    return response.json();
+                }).then(data =>
+                this.setState({
+                    logContent: data.inputStreamContent,
+                    isLogTruncated: data.truncated,
+                    inputStreamSize: data.completeInputStreamSize
+                }));
+        }
     }
-  }
 
-  handleLogDownloadErrorDialogClose = () => {
-    this.setState({
-      showLogDownloadErrorDialog: false,
-    });
-  };
+    handleLogDownloadErrorDialogClose = () => {
+        this.setState({
+            showLogDownloadErrorDialog: false,
+        });
+    };
 
-  handleLiveLogData(data) {
-    this.setState({ logContent: this.state.logContent + data });
-  }
+    handleLiveLogData(data) {
+        this.setState({ logContent: this.state.logContent + data });
+    }
 
-  handleError(response) {
-      if (response.status.toString() === HTTP_UNAUTHORIZED) {
-          window.location.replace(LOGIN_URI);
-      }
-      return response;
-  }
+    handleError(response) {
```

Comment:
Shall we go with two-space indentation for JS files?
JS and HTML files tend to have lot of inner elements. Four-space indentation makes it a bit harder to read.

@sameeraw @lasanthaDLPDS @vidurananayakkara wdyt?

Reply:
+1 Let's add two-space indentation for JS files.

Reply:
Cool.. Can you change and re-send?

Reply:
Fixed with 5c24430